frontend/src/app/user/service/user-email.service.ts

```diff
@@ -0,0 +1,26 @@
+import {Injectable} from '@angular/core';
+import {Router} from '@angular/router';
+
+@Injectable()
+export class UserEmailService {
+
+    localStorageKey = 'MyLittleUserEmailKey';
+
+    constructor(private router: Router) {
+    }
+
+    setEmail(email: string): void {
+        localStorage.setItem(this.localStorageKey, email);
+    }
+
+    getEmail(): string {
+        const stored = localStorage.getItem(this.localStorageKey);
+
+        // If we don't have an email, something is wrong.
+        if (!!stored) {
```

Comment:
If something is wrong, consider an alert message or console.log.

So we will know about it.